pingcap/tidb-binlog
# support TLS for components

### What problem does this PR solve? 
truely support TLS for components.
before this pr if enable TLS for components

- `tidb` will fail to connect to `pump`
- no TLS between drainer and pump
- no enable TLS for tikv client in `drainer`
- `binlogctl` can't work actually 
...

[relate docs](https://pingcap.com/docs/stable/how-to/secure/enable-tls-between-components/) ([Chinese version](https://pingcap.com/docs-cn/stable/how-to/secure/enable-tls-between-components/))
### What is changed and how it works?
- properly handle things about TLS when enabling TLS
- enable TLS in the integration tests
- log pump config at startup time

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)
enable TLS for all components and check replication works, include tidb/tikv/pd/pump/drainer/binlogctl

Code changes

Side effects

Related changes
 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository ???
having not tried using ansible to deploy.
 - Need to be included in the release note
